crowbar_framework/app/models/api/node.rb

```diff
@@ -136,9 +133,24 @@ def disable_pre_upgrade_attribute_for(name)
       true
     end
 
-    def save_node_state(message = "")
-      # FIXME: save the node status to global status
-      Rails.logger.info(message)
+    def save_node_state(role, state = "upgrading")
+      status = ::Crowbar::UpgradeStatus.new
+      status.save_current_node(
+        name: @node.name,
+        alias: @node.alias,
+        ip: @node.public_ip,
+        state: state,
+        role: role
+      )
+      if state == "upgraded"
+        progress = status.progress
+        remaining = progress[:remaining_nodes] - 1
+        upgraded = progress[:upgraded_nodes] + 1
+        ::Crowbar::UpgradeStatus.new.save_nodes(
+          upgraded, remaining < 0 ? 0 : remaining
```

Comment:
it looks like that, that's why i was asking. but why do we need this check then?